src/queue.js

```diff
@@ -1,5 +1,25 @@
 class Queue {
-
+  constructor() {
+    this.queue = {};
+    this.size = 0;
+    this.index = 0;
+  }
+  enqueue(item) {
+    this.queue[this.size] = item;
```

Comment:
I'm assuming that by doing this, your objects will all look like "associative arrays?" ex. ```{"0": "data", "1": "data2"}```

Reply:
If this is the case, why not try using Arrays as they're already inline with this type of associated behavior?